lib/pbench/server/api/resources/query_apis/metadata_index/timeseries_samples.py

```diff
@@ -0,0 +1,232 @@
+        results of the original query; otherwise we form a new query, either
+        with the specified filters or without any filters (returning up to 10000
+        documents at a time).
+
+        Args:
+            json_data:
+                "run_id": run id of a dataset that client wants to search for
+                        fetching the result-data document samples.
+
+                "scroll_id": Server provided Elasticsearch scroll id that client
+                            recieved in the result of the original query with
+                            filters. This can be used to fetch the next page of
+                            the result.
+
+                "filters": key-value representation of query filter parameters to
+                           narrow the search results e.g. {"sample.name": "sample1"}
+
+        EXAMPLES:
+            {
+                "run_id": "1234567"
+                "filters": {"sample.name": "sample1"},
+            }
+            or
+            {
+                "run_id": "1234567"
+                "scroll_id": "cmFuZG9tX3Njcm9sbF9pZF9zdHJpbmdfMg=="
+            }
+        """
+        run_id = context["run_id"]
+        dataset = context["dataset"]
+        scroll_id = json_data.get("scroll_id")
+
+        self.logger.info(
+            "Return TimeSeries sample rows {} for dataset {}, prefix {}, "
+            "run id {} next page "
+            if scroll_id
+            else "",
+            dataset,
+            self.prefix,
+            run_id,
+        )
+
+        scroll_id = json_data.get("scroll_id")
+        if scroll_id:
+            return {
+                "path": "/_search/scroll",
+                "kwargs": {
+                    "json": {
+                        "scroll": TimeSeriesSamples.SCROLL_EXPIRY,
+                        "scroll_id": scroll_id,
+                    }
+                },
+            }
+
+        # Retrieve the ES indices that belong to this run_id
+        indices = self.get_index(dataset, "result-data")
+        if not indices:
+            self.logger.debug(
+                f"Found no indices matching the prefix result-data"
+                f"for a dataset {dataset!r}"
+            )
+            abort(HTTPStatus.NOT_FOUND, message="Found no matching indices")
+
+        try:
+            # Retrive result-data mappings for validating user provided filters
+            mappings = self.get_mappings("result-data")
+        except (TemplateNotFound, KeyError) as e:
+            self.logger.exception(f"Exception while getting 'result-data' template {e}")
+            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
+
+        es_filter = [{"match": {"run.id": run_id}}]
+        query_strings = []
+
+        # Validate each user provided filters against the result-data document
+        # mappings. If the filter is not present in result-data mappings
+        # properties we will not apply that filter in the ES query.
```

Comment:
If the filter is not present, then we "fail" silently...I'm not sure I consider this a feature.  I suggest that either
- we validate the filters and return an error for any which aren't in the mappings; or
- we don't validate the filters and let ES deal with it (e.g., by returning an empty result...which I know that Dave won't support 😉).